### PR TITLE
Supoort -overwrite flag in signer

### PR DESCRIPTION
**Pull request by ooq**
A new bool flag `overwrite` in signer that allows overwriting existing attestation.

### CODE REVIEW

File: pkg/kritis/signer/signer.go [hunk 1]
```diff
@@ -78,23 +82,36 @@ var (
 // SignImage signs an image without doing any policy check.
 // Returns an error if creating an attestation fails.
 func (s Signer) SignImage(image string) error {
-	existed, _ := s.isAttestationAlreadyExist(image)
+	existed, err := s.isAttestationAlreadyExist(image)
+	if err != nil {
+		return fmt.Errorf("checking existing attestation status failed: %v", err)
+	}
 	if existed {
```
Comment (nenaddedic) [inline pkg/kritis/signer/signer.go:89]: Optional: I suggest restructuring the control flow to reduce nesting, bail out early, and make it easier to follow, maybe like this:
if exists && !overwrite {
  return ...
}
if exists && overwrite {
  delete ...
} else {
  doesn't exist
}
create attestation

File: integration/signer/tests/test-overwrite.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+docker build --no-cache -t $GOOD_IMAGE_URL -f ./Dockerfile.good .
+
+trap 'delete_image $GOOD_IMAGE_URL'  EXIT
+
+# push good image
+docker push $GOOD_IMAGE_URL
+# get image url with digest format
+GOOD_IMG_DIGEST_URL=$(docker image inspect $GOOD_IMAGE_URL --format '{{index .RepoDigests 0}}')
+
+trap 'delete_occ $GOOD_IMG_DIGEST_URL'  EXIT
```
Comment (nenaddedic) [inline integration/signer/tests/test-overwrite.sh:32]: Do traps "stack" like this?  At this point, will both delete_occ and delete_image be executed on EXIT?  Is the order specified?

(This is for my education...)

File: integration/signer/tests/test-overwrite.sh [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+# sign good image in bypass mode
```
Comment (nenaddedic) [inline integration/signer/tests/test-overwrite.sh:34]: Nit:
If it's bypassing policy checks, then "good image" is a bit misleading.  There's no good or bad images.

File: integration/signer/tests/test-overwrite.sh [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+-mode=bypass-and-sign \
+-image=${GOOD_IMG_DIGEST_URL} \
+-pgp_private_key=private.key \
+-policy=policy.yaml \
+-note_name=${NOTE_NAME}
+
+sleep 5
+
+# sign good image in bypass mode
+./signer -v 10 \
```
Comment (nenaddedic) [inline integration/signer/tests/test-overwrite.sh:46]: This just checks that the signing doesn't fail, right?  It doesn't actually verify that the attestation was overwritten, correct?

File: pkg/kritis/signer/signer.go (hunk 1 above)
Comment (ooq) [reply pkg/kritis/signer/signer.go]: Done.

File: integration/signer/tests/test-overwrite.sh (hunk 2 above)
Comment (ooq) [reply integration/signer/tests/test-overwrite.sh:32]: Actually your concern is valid. They don't stack...fixing them in #567 instead to reduce merge conflicts.

File: integration/signer/tests/test-overwrite.sh (hunk 3 above)
Comment (ooq) [reply integration/signer/tests/test-overwrite.sh:34]: Done.

File: integration/signer/tests/test-overwrite.sh (hunk 4 above)
Comment (ooq) [reply integration/signer/tests/test-overwrite.sh:49]: Added check to validate that occurrence id before and after overwrite are different.

**Comment by ooq** [comment]
@nenaddedic Addressed comments. PTAL.

File: integration/signer/tests/test-overwrite.sh (hunk 4 above)
Comment (nenaddedic) [reply integration/signer/tests/test-overwrite.sh:49]: Thanks for doing this!  And, sorry, my comment was an honest question, I didn't mean to insist that this needs to be tested so thoroughly.  Either way, this is great now!

**Pull request merged**